src/encoder/serde.rs

```diff
@@ -496,6 +496,17 @@ impl Serialize for TimeStamp {
     }
 }
 
+impl Serialize for RegExp {
+    #[inline]
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let doc = Bson::RegExp(self.clone());
```

Comment:
I know this is used elsewhere in this file, but this variable probably shouldn't be named `doc`.

Reply:
Yeah, I agree; I should have paid more attention to that. I'll update it in all of these implementations.